### PR TITLE
Cleaning up heading styles

### PR DESCRIPTION
**Pull request by noahmanger**
## Summary
This refactors our heading Sass to clean up cruft and simplify and unify the styles for page and section headings. Because headings can contain multiple elements — a title (as an `h1`, `h2`, etc.), and optionally a subtitle, an action, or a tag — instead of applying the border style to the `h*` elements, I made two new classes: `.heading--main` and `.heading--section`, which are applied to a `div` that wraps the whole set of elements. (I'm open to renaming these classes if there's other names that would be clearer). 

So the new structure for a page header, for example, is:
```
<div class="heading--main">
  <h1>Page heading</h1>
</div>
```

The stuff inside only needs a class if it's either a subtitle or if there's an action (buttons, selects, etc.) that needs to be floated. In that case, we just use `.heading__left` on the title and `.heading__right` on the button or whatever action.

Refactoring something like this is definitely tricky and it's easy to miss things, so I'm going to ask @jenniferthibault to take a careful look through. 

## Screenshots
In the styleguide:
image

In the wild:
image

image

image

image

cc @jenniferthibault @adborden 

Resolves https://github.com/18F/fec-style/issues/490
Requires https://github.com/18F/openFEC-web-app/pull/1592
Requires https://github.com/18F/fec-cms/pull/509

### CODE REVIEW

File: scss/components/_headings.scss [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+//  </div>
+//
+// Styleguide components.headings-with-action.1
+
+.heading--with-action {
+  h1,
+  h2,
+  h3,
+  h4 {
+    line-height: u(3.4rem);
```
Comment (adborden) [inline scss/components/_headings.scss]: Is this a well-known font-size or something? I thought it was best practice [not to specify units](https://pages.18f.gov/frontend/#measurements) for `line-height`.

**Comment by adborden** [comment]
:+1: awesome! I really like the additional documentation.

File: scss/components/_headings.scss (hunk 1 above)
Comment (noahmanger) [inline scss/components/_headings.scss]: Yeah, I agree not best-practice. This is kind of a hack in order to have the buttons and heading line up correctly, like:


Not sure if there's a better way

**Comment by jenniferthibault** [comment]
@noahmanger this. is awesome. I am so excited. I left a few comments above (mostly for my own clarity). The only thing I'm seeing as a possible error is a little funkiness on the Headings with dates style:

screen shot 2016-10-04 at 5 29 17 pm

File: scss/components/_headings.scss [hunk 2]
```diff
@@ -0,0 +1,124 @@
+// Headings
+//
+// Contain the title, date select and action button.
+// Use .heading--main when the heading is the top-level heading on a page as it provides extra margin-bottom
+// Use .heading--section When the heading is just a heading of a section on a page for less margin-bottom
```
Comment (noahmanger) [inline scss/components/_headings.scss]: Just figured out that we can use markdown formatting here to format, so I used the tic marks.

File: scss/components/_headings.scss [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  letter-spacing: -.3px;
+  margin-bottom: u(1rem);
+  text-transform: uppercase;
+}
+
+// Headings with actions
+//
+// When the heading title includes an action, add this class to one of the base classes.
+// Set line height to the height of the buttons or selects in the heading
+// Use .heading__left and .heading__right to float elements at $med breakpoints
```
Comment (noahmanger) [inline scss/components/_headings.scss]: Fixed when I figured out it's because of markdown formatting.

**Comment by noahmanger** [comment]
All good!

File: scss/components/_headings.scss (hunk 1 above)
Comment (adborden) [inline scss/components/_headings.scss]: :+1: let's just add a comment that the 3.4rem is coming from the size of the action button.

**Pull request merged**